public/app/core/utils/richHistory.ts

```diff
@@ -329,6 +329,55 @@ export function notEmptyQuery(query: DataQuery) {
   return false;
 }
 
+export function filterQueriesBySearchFilter(queries: RichHistoryQuery[], searchFilter: string) {
+  const filteredQueries = queries.filter(query => {
+    const searchInComment = query.comment.includes(searchFilter);
+    const searchInQueries = query.queries.filter(query => {
+      const queryHasSearchedValue = Object.values(_.omit(query, 'datasource')).some(value =>
```

Comment:
I think in addition to `datasource`, it would also be worthwhile to omit other internal props that definitely exist on every `DataQuery`, like `refId`, `key`, etc.

Reply:
Yes, updated and added comment in https://github.com/grafana/grafana/pull/25747/commits/3313c6b0e430ceeb85beb17670c8b1215518229a

Reply:
Unless you know of some reason why not to, I would argue that all properties of the base `DataQuery` type should be omitted, even e.g. `hide`. I don't know whether it's set on the queries in the query history but I find it conceivable that a user might e.g. mention "false" in their comments and search for it. ;) (And it's not much overhead to add it to the list of omitted props.)

Reply:
Yes. Great catch and point. I have added the rest of the `DataQuery` properties that we don't want to match.